ng-dev/misc/new-main-branch/remote-fork-update.ts

```diff
@@ -0,0 +1,92 @@
+/**
+ * @license
+ * Copyright Google LLC All Rights Reserved.
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {debug, green, promptConfirm, promptInput, warn, yellow} from '../../utils/console';
+
+import {AuthenticatedGitClient} from '../../utils/git/authenticated-git-client';
+import {GithubRepo} from '../../utils/git/github';
+import {info} from 'console';
+
+/**
+ * Prompts the user whether remote forks for the current repository should be
+ * updated from `master` to `main`.
+ *
+ * An access token for performing the Github Admin operation is demanded through
+ * a prompt. This is opt-in as not every contributor may want to grant the tool a
+ * Github access token/or some contributors may want to make the changes themselves.
+ */
+export async function promptForRemoteForkUpdate() {
+  if (
+    !(await promptConfirm(
+      'Do you want to update your forks on Github to also use `main`? (recommended)',
+    ))
+  ) {
+    return;
+  }
+
+  info('');
+  info(yellow('In order to be able to update your fork automatically, the script needs'));
+  info(yellow('access to your GitHub account. A token needs to be provided in the next steps.'));
```

Comment:
We need to express that the token is used in memory here and not stored or sent anywhere else.

It also might be better to say that the script needs to make authenticated requests to github for you, the script is doing the actions on behalf of the user.  It might be a little clearer and less scary for users.

Reply:
updated